pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -54,12 +54,8 @@ func AddToManager(mgr manager.Manager,
 		r.registryClients[clusterName] = imagestreamtagwrapper.MustNew(m.GetClient(), m.GetCache())
 	}
 	c, err := controller.New(ControllerName, mgr, controller.Options{
-		Reconciler: r,
-		// We conflict on ImageStream level which means multiple request for imagestreamtags
-		// of the same imagestream will conflict so stay at one worker in order to reduce the
-		// number of errors we see. If we hit performance issues, we will probably need cluster
-		// and/or imagestream level locking.
-		MaxConcurrentReconciles: 1,
+		Reconciler:              r,
+		MaxConcurrentReconciles: 100,
```

Comment:
You need to synchronize this on imagestream level to prevent races

Reply:
i.E. at any time, at most one worker can run per imagestream

Reply:
See the comment you removed for an explanation

Reply:
/hold

Reply:
Stole the code from https://github.com/kubernetes/test-infra/blob/dbc92a6558ab6ae3fc8ef5d10a2bb629f8f0d26a/prow/crier/reporters/github/reporter.go#L141-L149